pkg/api/v1/defaults.go

```diff
@@ -101,6 +102,9 @@ func SetDefaults_Service(obj *v1.Service) {
 	if obj.Spec.SessionAffinity == "" {
 		obj.Spec.SessionAffinity = v1.ServiceAffinityNone
 	}
+	if obj.Spec.SessionAffinity == v1.ServiceAffinityClientIP && obj.Spec.ClientIPAffinityConfig == nil {
+		defaultServiceAffinityConfig(obj.Spec.ClientIPAffinityConfig)
```

Comment:
Did you test this?  It can't possibly work.  You're passing `nil` to the function and then changing it inside teh function, but not passing the result back.

There's no need for a function here.  It's called once.  Just inline it.  And test it.